serverlessworkflow/modules/ROOT/pages/testing-and-troubleshooting/integration-tests-with-postgresql.adoc

```diff
@@ -1 +1,198 @@
+{quarkus_testing_guide_test_resource}[Starting services before the Quarkus application starts] guide.
+* *Kogito-quarkus-test-utils* with some Kogito predefined testcontainers as the PostgreSQL one.
+
+The testing case followed in this document is based on the `serverless-workflow-examples/serverless-workflow-newsletter-subscription` example application.
+You can access this example application in {kogito_sw_examples_git_repo_url}[Kogito Examples] GitHub repository
+
+
+[[creating-integration-test]]
+== Creating the integration test with postgreSQL
+Let's create the integration test for a specific Serverless Workflow with PostgresQL:
+
+First, we need to include the dependency to kogito-quarkus-test-utils at the project pom.xml
+--
+
+.Dependencies in `pom.xml` file
+[source,xml]
+----
+<dependency>
+  <groupId>org.kie.kogito</groupId>
+  <artifactId>kogito-quarkus-test-utils</artifactId>
+  <scope>test</scope>
+</dependency>
+----
+--
+This allows us to use predefined test resources such the one we need in this case for PostgreSQL: PostgreSqlQuarkusTestResource
+
+[NOTE]
+====
+This test framework allows the use of predefined test resources like: KafkaQuarkusTestResource, KeycloakQuarkusTestResource that can be used to test
+more complex scenarios
+====
+
+The next step would be to add the test annotations to the integration test resources
+
+.Example integration test annotation
+[source,java]
+----
+@QuarkusIntegrationTest <1>
+@QuarkusTestResource(PostgreSqlQuarkusTestResource.class) <2>
+class HelloTestIT {<3>
+
+}
+----
+
+<1> Allows launch and test the artifact produced by the Quarkus build. Supports testing a jar (of whichever type), a native image or container image
+<2> Add the reference to the predefined PostgreSqlQuarkusTestResource in kogito-quarkus-test-utils. The annotation
+@QuarkusTestResource makes Quarkus run the corresponding QuarkusTestResourceLifecycleManager before any tests are run.
+<3> The naming of the test would end with 'IT' to identify which test needs to be executed as an integration test.
+
+Once the needed resources and annotation are specified, We can start testing the different interactions with the Serverless Workflow like it
+ is explained in {basic-integration-test-with-restassured}[Testing your Serverless Workflow application using REST Assured].
+
+[[Add-postgreSQL-test-resource]]
+== PostgreSQL instance as test resource
+The test are using PostgreSqlQuarkusTestResource to start the PostgresSQL instance. PostgreSqlQuarkusTestResource is a predefined test
+ resource provided by kogito-quarkus-test-utils that starts a container with the PostgreSQL instance. That container has a full set of
+ default configuration that are included in the integration test automatically
+
+[NOTE]
+====
+As a test resource, it will run its QuarkusTestResourceLifecycleManager before the service tests are executed.
+More related info in {quarkus_testing_guide_test_resource}[TESTING YOUR APPLICATION] Quarkus guide.
+====
```

Comment:
Is there a direct anchor/link to the class `QuarkusTestResourceLifecycleManager` in this guide you linked here? If so, what do you think about adding it here?